hotshot-state-prover/src/service.rs

```diff
@@ -382,19 +354,34 @@ pub async fn sync_state<ApiVer: StaticVersionType>(
     tracing::debug!("Old state: {old_state:?}");
     tracing::debug!("New state: {:?}", bundle.state);
 
+    let blocks_per_epoch = config
+        .blocks_per_epoch()
+        .await
+        .map_err(ProverError::NetworkError)?;
+    let next_stake = if is_last_block(bundle.state.block_height as u64, blocks_per_epoch) {
```

Comment:
nit: The last block of an epoch should be epoch root. Because, only epoch root is persisted in the consensus and the stake table update is bundled with it. 
I can also fix this in my later PR to fetch epoch root information from the hotshot query service.

Reply:
issue: https://github.com/EspressoSystems/espresso-network/issues/2924